### PR TITLE
docs: Add SSL to documentation

### PR DESCRIPTION
**Pull request by yangobeil**
In this PR I add the new functions related to semi supervised learning to the documentation and I add an example of how to use them. I also update the package version to 1.2.0 in order to release the new version on pip.

The only files that need reviewing for the docs are the .rst files. You can also visualize the results by cloning the branch and opening the docs/_build/html/index.html file.

### CODE REVIEW

File: docs/data_example.rst [hunk 1]
```diff
@@ -6,8 +6,8 @@ perform data augmentation on your images and transform the dataset into tfrecord
 The data can come from any source and the only requirement is that the images are saved in separate folders for each class.
 
 
-Spliting a set of images into a training and a validation set
--------------------------------------------------------------
+Splitting a set of images into a training and a validation set
+----------------------------------------------------------------
 
 If you have a single set of images, but would like to split it into a training and a validation set, first place your images in 
 a train folder, which is located in the folder 'data/image_dataset' for this example. Using hockey and soccer players as examples
```
Comment (hasibzunair) [inline docs/data_example.rst]: Maybe we can say "located in the folder `data/image_dataset`. For this example, we use hockey and soccer players as examples"?

File: docs/index.rst [hunk 2]
```diff
@@ -15,10 +15,17 @@ built on top of various pretrained models, optimize model hyperparameters using
 the accuracy on a test set. The library capitalizes on the concepts of data augmentation, fine tuning and 
 hyperparameter optimization, to achieve high accuracy given small sets of training images.
 
-This library has a few distinguishing features. It is specifically designed to work with Google Colab notebooks and leverage their TPUs, to seamlessly transition from quick iterations of modeling approaches to large-scale training. Functionalities of hyperparmaters tuning and progressive learning are included and easily integrated in the pipeline to reach higher accuracy. Finally, state-of-the art efficientnet transfer learning models and Tensorflow 2 functionalities are considered to provide high efficiency. 
+This library has a few distinguishing features. It is specifically designed to work with Google Colab notebooks and
+leverage their TPUs, to seamlessly transition from quick iterations of modeling approaches to large-scale training.
+Functionalities of hyperparmaters tuning and progressive learning are included and easily integrated in the pipeline to
+reach higher accuracy. Finally, state-of-the art efficientnet transfer learning models and Tensorflow 2 functionalities
```
Comment (hasibzunair) [inline docs/index.rst]: EfficientNet instead of efficientnet. TensorFlow instead of Tensorflow.

File: docs/index.rst [hunk 3]
```diff
@@ -15,10 +15,17 @@ built on top of various pretrained models, optimize model hyperparameters using
...
+are considered to provide high efficiency.
 
 A great way of explaining the library is using an example notebook, which you can find `here <https://colab.research.google.com/drive/1s9pnNotdPoHI4n_7OZl4zwEP9rccJ3f-?usp=sharing/>`_. 
 
+The most recent version of this library adds a feature to leverage unlabelled images in order to improve the performance
+of image classifiers. This procedure is called semi-supervised learning and is discussed in this `blog post <https://medium.com/decathlondevelopers/improving-performance-of-image-classification-models-using-pretraining-and-a-combination-of-e271c96808d2/>`_.
```
Comment (hasibzunair) [inline docs/index.rst]: I would write semi-supervised learning (SSL).

File: docs/ssl_example.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
+Improve a model using unlabelled images
```
Comment (hasibzunair) [inline docs/ssl_example.rst]: Maybe write classification model to be specific?

File: docs/ssl_example.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+========================================
+
+This code example shows how to use this library to exploit unlabelled data to increase the performance of an image
+classifier. The main procedure consists in first training a classifier using the labelled data (or using an already
+trained one). Then this model is used to make predictions for the unlabelled data, which are used as pseudo labels to
+create a new dataset. This larger dataset is then used to train a larger and more performant classifier.
```
Comment (hasibzunair) [inline docs/ssl_example.rst]: create a new dataset which consists of both the pseudo labels and the training data previously used to train the classifier.

File: docs/ssl_example.rst [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+following code::
+
+  generator = decavision.dataset_preparation.generate_pseudolabels.PseudoLabelGenerator()
+  generator.generate_pseudolabel_data()
+
+This uses the teacher model to predict the label of each unlabelled image. These predictions are then saved in a csv file at 'outputs/data.csv'.
+
+Since the unlabelled images can come from different places, it is important to make sure that bad images are not used
+further. To ensure data quality, it can be helpful to plot the distribution of the predictions using::
+
```
Comment (hasibzunair) [inline docs/ssl_example.rst:36]: Two colons.

File: docs/ssl_example.rst [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  generator.plot_confidence_scores(per_class=False)
+
+A chart will be saved in the 'outputs' folder with the highest probability predicted for each image. If some probabilities
+are too low, it means that the image is so bad that the teacher model does not recognize it. Such image should ideally
+be discarded. This chart thus helps pick the threshold variable that will be used to create the new dataset with only relevant images.
+
+The final step is to use the predictions along with the threshold (if one is used) to distribute the unlabelled images
+into classes. This pseudo labelled dataset is then combined with the original training data to make a larger dataset::
+
```
Comment (hasibzunair) [inline docs/ssl_example.rst:45]: Two colons.

**Review by hasibzunair — COMMENTED** [review]
Some minor edits!

**Comment by hasibzunair** [comment]
I think the PR title needs to be changed in order to remove the check warning. Options are listed here haha https://github.com/commitizen/conventional-commit-types/blob/master/index.json

File: docs/ssl_example.rst (hunk 6 above)
Comment (yangobeil) [reply docs/ssl_example.rst:36]: Actually it's ok like that. It is the format used for rst files.

File: docs/index.rst (hunk 3 above)
Comment (yangobeil) [reply docs/index.rst]: Good idea!

**Comment by yangobeil** [comment]
> I think the PR title needs to be changed in order to remove the check warning. Options are listed here haha https://github.com/commitizen/conventional-commit-types/blob/master/index.json

Indeed I totally forgot about that haha thanks!

File: docs/ssl_example.rst (hunk 6 above)
Comment (hasibzunair) [reply docs/ssl_example.rst:36]: Oh okay!

**Review by hasibzunair — APPROVED** [review]
I was waiting to write this. LGTM!
